providers/digests.inc

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/*
+ * The including source file is expected to define ALG() and ALGC() in a
+ * way that's appropriate for them.
+ */
+
+#ifndef ALG
+# error "Undefined macro: ALG()"
+#endif
+
+#if defined(DEFAULT_PROVIDER) || defined(FIPS_PROVIDER)
+    /* Our primary name:NIST name[:our older names][:OID] */
+    ALG("SHA1:SHA-1:SSL3-SHA1:1.3.14.3.2.26", ossl_sha1_functions),
+    ALG("SHA2-224:SHA-224:SHA224:2.16.840.1.101.3.4.2.4",
+        ossl_sha224_functions),
+    ALG("SHA2-256:SHA-256:SHA256:2.16.840.1.101.3.4.2.1",
+        ossl_sha256_functions),
+    ALG("SHA2-384:SHA-384:SHA384:2.16.840.1.101.3.4.2.2",
+        ossl_sha384_functions),
+    ALG("SHA2-512:SHA-512:SHA512:2.16.840.1.101.3.4.2.3",
+        ossl_sha512_functions),
+    ALG("SHA2-512/224:SHA-512/224:SHA512-224:2.16.840.1.101.3.4.2.5",
+        ossl_sha512_224_functions),
+    ALG("SHA2-512/256:SHA-512/256:SHA512-256:2.16.840.1.101.3.4.2.6",
+        ossl_sha512_256_functions),
+
+    /* We agree with NIST here, so one name only */
+    ALG("SHA3-224:2.16.840.1.101.3.4.2.7", ossl_sha3_224_functions),
+    ALG("SHA3-256:2.16.840.1.101.3.4.2.8", ossl_sha3_256_functions),
+    ALG("SHA3-384:2.16.840.1.101.3.4.2.9", ossl_sha3_384_functions),
+    ALG("SHA3-512:2.16.840.1.101.3.4.2.10", ossl_sha3_512_functions),
+
+    ALG("SHAKE-128:SHAKE128:2.16.840.1.101.3.4.2.11",
+        ossl_shake_128_functions),
+    ALG("SHAKE-256:SHAKE256:2.16.840.1.101.3.4.2.12",
+        ossl_shake_256_functions),
+
+    /*
+     * KECCAK-KMAC-128 and KECCAK-KMAC-256 as hashes are mostly useful for
+     * KMAC128 and KMAC256.
+     */
+    ALG("KECCAK-KMAC-128:KECCAK-KMAC128", ossl_keccak_kmac_128_functions),
+    ALG("KECCAK-KMAC-256:KECCAK-KMAC256", ossl_keccak_kmac_256_functions),
+#endif
+
+#ifdef DEFAULT_PROVIDER
+    /* Our primary name:NIST name[:OID] */
+    ALG("SHAKE-128:SHAKE128:2.16.840.1.101.3.4.2.11", ossl_shake_128_functions),
```

Comment:
Shouldn't these two be in the FIPS provider?

Reply:
Yup.  And they are, these are accidental duplicates.  Thanks for noticing